Source/WebKit/UIProcess/Cocoa/GPUProcessProxyCocoa.mm

```diff
@@ -89,6 +95,53 @@
     return SandboxExtension::createHandlesForMachLookup({ "com.apple.gputools.service"_s, }, std::nullopt);
 }
 
+RefPtr<XPCEventHandler> GPUProcessProxy::xpcEventHandler() const
+{
+    return adoptRef(new GPUProcessProxy::XPCEventHandler(*this));
+}
+
+bool GPUProcessProxy::XPCEventHandler::handleXPCEvent(xpc_object_t event) const
+{
+    if (!m_gpuProcess)
+        return false;
+
+    if (!event || xpc_get_type(event) == XPC_TYPE_ERROR)
```

Comment:
Should we be logging when we receive an XPC error?

Reply:
That is a good point. I will add logging in the upcoming patch.